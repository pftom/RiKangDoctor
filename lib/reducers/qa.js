Object.defineProperty(exports,"__esModule",{value:true});var _immutable=require('immutable');


var _constants=require('../constants/');

















var initialQaValue=(0,_immutable.Map)({
questions:null,
question:null,
AllImg:null,
isLoadingData:false,
loadingError:false,
loadingSuccess:false});


var qa=function qa(){var state=arguments.length>0&&arguments[0]!==undefined?arguments[0]:initialQaValue;var action=arguments[1];
switch(action.type){
case _constants.GET_QUESTIONS:
case _constants.GET_SINGLE_QUESTION:
case _constants.GET_SINGLE_QUESTION_ALL_IMG:

return state.merge({
isLoadingData:true,
loadingError:false,
loadingSuccess:false});


case _constants.GET_QUESTIONS_SUCCESS:var

questions=action.questions;

if(!state.get('questions')){
return state.merge({
questions:questions});

}

var oldQuestions=state.getIn(['questions','results']);
return state.
merge({
isLoadingData:false,
loadingSuccess:true,
questions:questions}).

updateIn(['questions','results'],function(list){return list.concat(oldQuestions);});


case _constants.GET_SINGLE_QUESTION_SUCCESS:var

question=action.question;

return state.merge({
isLoadingData:false,
loadingSuccess:true,
question:question});



case _constants.GET_SINGLE_QUESTION_ALL_IMG_SUCCESS:var

AllImg=action.AllImg;

return state.merge({
isLoadingData:false,
loadingSuccess:true,
AllImg:AllImg});


case _constants.GET_QUESTIONS_ERROR:
case _constants.GET_SINGLE_QUESTION_ERROR:
case _constants.GET_SINGLE_QUESTION_ALL_IMG_ERROR:

return state.merge({
isLoadingData:false,
loadingError:true});


default:
return state;}

};exports.default=

qa;