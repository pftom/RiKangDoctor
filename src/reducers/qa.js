import { List, Map } from 'immutable';

//import action constants
import { 
  GET_QUESTIONS,
  GET_QUESTIONS_SUCCESS,
  GET_QUESTIONS_ERROR,

  GET_SINGLE_QUESTION,
  GET_SINGLE_QUESTION_SUCCESS,
  GET_SINGLE_QUESTION_ERROR,

  GET_SINGLE_QUESTION_ALL_IMG,
  GET_SINGLE_QUESTION_ALL_IMG_SUCCESS,
  GET_SINGLE_QUESTION_ALL_IMG_ERROR,

} from '../constants/';

//import handle data func
import {
  combine,
  refreshIt,
} from './utils/';


//qa reducers
const initialQaValue = Map({
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});

const qa = (state = initialQaValue, action) => {
  switch (action.type) {
    case GET_QUESTIONS:
    case GET_SINGLE_QUESTION:
    case GET_SINGLE_QUESTION_ALL_IMG:

      return state.merge({
        isLoadingData: true,
        loadingError: false,
        loadingSuccess: false,
      });
    
    case GET_QUESTIONS_SUCCESS:


    const { questions, refresh } = action.payload;

      let oldQuestions = state.get('questions');

      if (questions) {
        questions = Immutable.Map(questions);
      }

      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        questions: refresh ? refreshIt(oldQuestions, questions) : combine(oldQuestions, questions),
      });

      
    case GET_SINGLE_QUESTION_SUCCESS:

      const { question } = action;
    
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        question,
      });

    
    case GET_SINGLE_QUESTION_ALL_IMG_SUCCESS:

      const { AllImg } = action;

      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        AllImg,
      }) 

    case GET_QUESTIONS_ERROR:
    case GET_SINGLE_QUESTION_ERROR:
    case GET_SINGLE_QUESTION_ALL_IMG_ERROR:

      return state.merge({
        isLoadingData: false,
        loadingError: true,
      });

    default:
      return state;
  }
};

export default qa;