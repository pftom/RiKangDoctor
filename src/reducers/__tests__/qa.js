import { Map, List } from 'immutable';

import qa from '../qa';

//import single action constants
import {
  GET_QUESTIONS,
  GET_QUESTIONS_SUCCESS,
  GET_QUESTIONS_ERROR,

  GET_SINGLE_QUESTION,
  GET_SINGLE_QUESTION_SUCCESS,
  GET_SINGLE_QUESTION_ERROR,

  GET_SINGLE_QUESTION_ALL_IMG,
  GET_SINGLE_QUESTION_ALL_IMG_SUCCESS,
  GET_SINGLE_QUESTION_ALL_IMG_ERROR,
} from '../../constants/';


//test questions
const getQuestionsAction = {
  type: GET_QUESTIONS,
};

const getQuestionsSuccessAction = {
  type: GET_QUESTIONS_SUCCESS,
  questions: {},
};

const getQuestionsErrorAction = {
  type: GET_QUESTIONS_ERROR,
};

//test single question
const getSingleQuestionAction = {
  type: GET_SINGLE_QUESTION,
};

const getSingleQuestionSuccessAction = {
  type: GET_SINGLE_QUESTION_SUCCESS,
  question: {},
};

const getSingleQuestionErrorAction = {
  type: GET_SINGLE_QUESTION_ERROR,
};

//test all img
const getSingleQuestionAllImgAction = {
  type: GET_SINGLE_QUESTION_ALL_IMG,
};

const getSingleQuestionAllImgSuccessAction = {
  type: GET_SINGLE_QUESTION_ALL_IMG_SUCCESS,
  AllImg: {},
};

const getSingleQuestionAllImgErrorAction = {
  type: GET_SINGLE_QUESTION_ALL_IMG_ERROR,
};

//qa reducers
const initialQaValue = Map({
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});


test('the qa reducer work as well', () => {
  //test questions
  expect(qa(initialQaValue, getQuestionsAction)).toMatchSnapshot();
  expect(qa(initialQaValue.merge({ isLoadingData: true }), getQuestionsSuccessAction)).toMatchSnapshot();
  expect(qa(initialQaValue.merge({ isLoadingData: true }), getQuestionsErrorAction)).toMatchSnapshot();

  //test single question
  expect(qa(initialQaValue, getSingleQuestionAction)).toMatchSnapshot();
  expect(qa(initialQaValue.merge({ isLoadingData: true }), getSingleQuestionSuccessAction)).toMatchSnapshot();
  expect(qa(initialQaValue.merge({ isLoadingData: true }), getSingleQuestionErrorAction)).toMatchSnapshot();

  //test all img
  expect(qa(initialQaValue, getSingleQuestionAction)).toMatchSnapshot();
  expect(qa(initialQaValue.merge({ isLoadingData: true }), getSingleQuestionAllImgSuccessAction)).toMatchSnapshot();
  expect(qa(initialQaValue.merge({ isLoadingData: true }), getSingleQuestionAllImgErrorAction)).toMatchSnapshot();
});