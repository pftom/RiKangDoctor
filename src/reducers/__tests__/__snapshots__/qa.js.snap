// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the qa reducer work as well 1`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the qa reducer work as well 2`] = `
Immutable.Map {
  questions: Immutable.Map {
  },
  question: null,
  AllImg: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the qa reducer work as well 3`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the qa reducer work as well 4`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the qa reducer work as well 5`] = `
Immutable.Map {
  questions: null,
  question: Immutable.Map {
  },
  AllImg: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the qa reducer work as well 6`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the qa reducer work as well 7`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the qa reducer work as well 8`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: Immutable.Map {
  },
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the qa reducer work as well 9`] = `
Immutable.Map {
  questions: null,
  question: null,
  AllImg: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;
